src/components/CompareResults/CompareResultsTable.tsx

```diff
@@ -6,52 +10,237 @@ import TableContainer from '@mui/material/TableContainer';
 import { useAppSelector } from '../../hooks/app';
 import type { CompareResultsItem } from '../../types/state';
-import type { CompareResultsTableHeaders } from '../../types/types';
+import type {
+  Platform,
+  CompareResultsTableHeader,
+  ConfidenceValue,
+  ActiveFilters,
+} from '../../types/types';
 import CompareResultsTableRow from './CompareResultsTableRow';
+import FilterOptionsList from './FilterOptionsList';
 
-const tableHeaders: CompareResultsTableHeaders[] = [
-  { id: 'platform', label: 'Platform', align: 'center' },
-  { id: 'graph', label: 'Graph', align: 'center' },
-  { id: 'test-name', label: 'Test Name', align: 'left' },
-  { id: 'base-value', label: 'Base', align: 'center' },
-  { id: 'new-value', label: 'New', align: 'center' },
-  { id: 'delta-percent', label: 'Delta', align: 'center' },
-  { id: 'confidence', label: 'Confidence', align: 'center' },
-  { id: 'total-runs', label: 'Total Runs', align: 'center' },
+const tableHead: CompareResultsTableHeader[] = [
+  {
+    id: 'platform',
+    label: 'Platform',
+    key: 'platform',
+    align: 'center',
+  },
+  {
+    id: 'graph',
+    label: 'Graph',
+    key: 'graph',
+    align: 'center',
+  },
+  {
+    id: 'test-name',
+    label: 'Test Name',
+    key: 'test',
+    align: 'left',
+  },
+  {
+    id: 'base-value',
+    label: 'Base',
+    key: 'base',
+    align: 'center',
+  },
+  {
+    id: 'new-value',
+    label: 'New',
+    key: 'new',
+    align: 'center',
+  },
+  {
+    id: 'delta-percent',
+    label: 'Delta',
+    key: 'delta',
+    align: 'center',
+  },
+  {
+    id: 'confidence',
+    label: 'Confidence',
+    key: 'confidence',
+    align: 'center',
+  },
+  {
+    id: 'total-runs',
+    label: 'Total Runs',
+    key: 'run',
+    align: 'center',
+  },
 ];
 
+interface Filter {
+  platform: {
+    options: Set<Platform>;
+    active: Platform[] | null;
+  };
+  test: {
+    options: Set<string>;
+    active: string[] | null;
+  };
+  confidence: {
+    options: Set<ConfidenceValue>;
+    active: ConfidenceValue[] | null;
+  };
+}
+
+const filter: Filter = {
+  platform: {
+    options: new Set([]),
+    active: null,
+  },
+  test: {
+    options: new Set([]),
+    active: null,
+  },
+  confidence: {
+    options: new Set(['Low', 'Medium', 'High'] as ConfidenceValue[]),
+    active: null,
+  },
+};
+
 function CompareResultsTable(props: CompareResultsProps) {
   const { mode } = props;
   const compareResults: CompareResultsItem[] = useAppSelector(
     (state) => state.compareResults,
   );
+  const filteredResults: CompareResultsItem[] = useAppSelector(
+    (state: RootState) => state.filterCompareResults.filteredResults,
+  );
+
+  filter.platform.options = new Set(
+    compareResults.map((result) => result.platform),
+  );
+
+  const suiteOptions = compareResults.map((result) => result.suite);
+  filter.test.options = new Set([
+    ...compareResults.map((result) => result.test),
+    ...suiteOptions,
+  ]);
```

Comment:
instead of changing the filter object above, I think I'd create the full filter object at each render. (in the extracted component as explained below). But I'm not so sure.
`active` doesn't seem used BTW, so maybe you can simplify the object:
```js
const filterOptions = {
  platform: new Set(...),
  test: ...,
  confidence: ...
};
```